app/views/vm_common/console_webmks.html.haml

```diff
@@ -14,7 +14,16 @@
             console.log("connection state change: connected");
           }
         });
-        wmks.connect("#{j webmks_uri.to_s}");
+
+        var host = window.location.hostname;
+        var encrypt = window.location.protocol === 'https:';
+        var port = encrypt ? 443 : 80;
+        if (window.location.port) {
+          port = window.location.port;
+        }
+        var proto = encrypt ? 'wss' : 'ws'
+
+        wmks.connect(proto + "://" + host + ":" + port + "/#{@console[:url]}");
```

Comment:
can you please add a test that shows that the url is generated correctly?

Reply:
@ohadlevy not sure if it's possible to test inline JS without a browser. I know this piece of code should be in a JS file and included, but as I wrote in the PR description, this is going to be backported and I would like to minimize the number of changes. I would like to create a followup PR where I move the JS to an external file (that is testable) similarly to the VNC/SPICE implementation, so the view can be replaced by the [universal one](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/views/layouts/remote_console.html.haml#L9). This followup PR would not be backported and it will contain the tests you're requesting for all console types. Is that OK?

Reply:
overall :+1: you might be able to do a simple functional test and use a regexp - but solving this correctly is probably more important :)

General comment is that  we should not keep on creating code which is hard to test... thanks!

Reply:
@ohadlevy : How do you test the websocket javascript stuff in Foreman? @skateman : Maybe we can get some inspiration from there.

Reply:
I'd be very interested in having some testing around all the websocket connections and consoles.